### PR TITLE
LPS-156273 Add dropdown support for DM extensions

### PR DESCRIPTION
**Pull request by adolfopa**
@ambrinchaudhary I've add the backend code to create all the dropdowns; what is remaining is moving all actions to a props transformer.

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
@adolfopa I've been thinking... why is it redirecting to google docs /one drive with "post"? we are not sending anything. Shouldn't be simple links? Maybe "post" was set by mistake?

**Comment by boton** [comment]
@ambrinchaudhary it's right I try to use a POST instead of `href` but is working as a GET in the old code too.

The old code specifies [POST navigation](https://github.com/liferay/liferay-portal/blob/11724df470f8b261c94043e706e8a8f438a8e4c6/modules/apps/document-library-opener/document-library-opener-google-drive-web/src/main/java/com/liferay/document/library/opener/google/drive/web/internal/display/context/DLOpenerGoogleDriveDLViewFileVersionDisplayContext.java#L170) but later on the [IconTag](https://github.com/liferay/liferay-portal/blob/de0a95d28e8191c1884418c564eead3ad9f0cf97/util-taglib/src/com/liferay/taglib/ui/IconTag.java#L360-L365) our method is "POST" isn't equal to "post" and doesn't work like a POST (only making a post with SPA activated, like the new href). Is this a bug? I do not know

After that, I only used a `setHref` and it's working.

@adolfopa I'm going to need your help here 12b0b3a, I assumed the edit item is inside of the first element is a group, and adding it at the end of the first group if not but maybe we can do it more safely.

I'm not able to check it in Office because the team credentials aren't working

```
2022-06-28 13:01:37.652 ERROR [http-nio-8080-exec-2][OneDriveOAuth2Servlet:115] null
com.github.scribejava.core.model.OAuth2AccessTokenErrorResponse: {"error":"invalid_client","error_description":"AADSTS7000222: The provided client secret keys for app 'xxx-xx-xx-xx-xxxxx' are expired. Visit the Azure portal to create new keys for your app: https://aka.ms/NewClientSecret, or consider using certificate credentials for added security: https://aka.ms/certCreds.\r\nTrace ID:  ....,"error_uri":"https://login.microsoftonline.com/error?code=7000222"}
```

**Comment by adolfopa** [comment]
Just started reviewing :)

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/display/context/DLOpenerOneDriveDLViewFileVersionDisplayContext.java [hunk 1]
```diff
@@ -165,6 +205,30 @@ public List<ToolbarItem> getToolbarItems() throws PortalException {
 		return toolbarItems;
 	}
 
+	private List<DropdownItem> _addEditInOffice365DropdownItem(
+		List<DropdownItem> dropdownItems,
+		DropdownItem editInOffice365DropdownItem) {
+
+		int i = 1;
+
+		for (DropdownItem dropdownItem : dropdownItems) {
```
Comment (AliciaGarciaGarcia) [inline modules/dxp/apps/document-library-opener/document-library-opener-onedrive-web/src/main/java/com/liferay/document/library/opener/onedrive/web/internal/display/context/DLOpenerOneDriveDLViewFileVersionDisplayContext.java:214]: If the dropdownItem is a group the option is added at the end of the dropdown instead down the edit option as before

image

but with this PR it shows at the final

image

@adolfopa I'll send 2 commits to make this, please review it

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by AliciaGarciaGarcia** [comment]
@adolfopa I added 2 commits as said on my review.

please review or delete them as you wish

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@AliciaGarciaGarcia See my changes.
@boton There are frontend errors when checking in a document. To reproduce:
1. Upload a TXT file to document and media.
2. Edit it in Google Docs; make an arbitrary change but don't save yet.
3. Go back to documents and media.
4. Execute "check in" in the TXT file.
This should bring back the changes from Google Docs, but it is complaining about a missing action.

**Comment by boton** [comment]
ok @adolfopa I'm going take a look

**Comment by boton** [comment]
> @boton There are frontend errors when checking in a document. To reproduce:
> 
> 1. Upload a TXT file to document and media.
> 1. Edit it in Google Docs; make an arbitrary change but don't save yet.
> 1. Go back to documents and media.
> 1. Execute "check in" in the TXT file.
> 1. This should bring back the changes from Google Docs, but it is complaining about a missing action.

@adolfopa I updated some things

- Fixed CHECKIN c0dc7a3b9136eee9683d2d06978883551f4d0bbb
- Fixed the CANCEL_CHECKOUT 8184f6f
- Remove unwanted data on CHECKIN ddf5df172ea6847d72cf179dbf5d6ab4ec30a3bf
  image
- Update the editOffice action a921c54
- Apply the same in OneDrive CHECKIN and CANCEL_CHECKOUT (not checked because credentials) 6eb1728
- Add support to dropdownItems outside groups 86d8cec

**Comment by adolfopa** [comment]
@boton Rebased, fixed some java issues and squashed. Will put on hold until we have the OneDrive issues sorted.

**Comment by Hanlf** [comment]
Hi @adolfopa , 
Is this PR blocked by OneDrive credentials? The credentials our QA use for CI works fine, see https://github.com/liferay-lima/liferay-portal/pull/3217#issuecomment-1176962383
Let us know if QA can help.

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:onedrive

**Comment by Hanlf** [comment]
ci:test:onedrive

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @adolfopa @ambrinchaudhary and @boton 
The functional test failure is a flaky issue, not related to this PR.
The modules-semantic-versioning-jdk8/0/0 failure should be related to this PR, could you take a look?
```
     [exec] > Task :apps:frontend-taglib:frontend-taglib-clay:baseline
     [exec]   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec] = ================================================== ========== ========== ========== ========== ==========
     [exec] * com.liferay.frontend.taglib.clay.data.set.servlet.taglib.util MINOR      1.2.0      1.2.0      1.3.0      VERSION INCREASE REQUIRED
     [exec] 	<   class      com.liferay.frontend.taglib.clay.data.set.servlet.taglib.util.ClayDataSetActionDropdownItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] * com.liferay.frontend.taglib.clay.servlet.taglib.util MAJOR      3.11.0     3.11.0     4.0.0      VERSION INCREASE REQUIRED
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownCheckboxItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownGroupItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$DropdownItemStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownRadioGroupItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] 	<   class      com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownRadioItem
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   return     void
     [exec] 	>   interface  com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterHrefStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$KeyStep
     [exec] 		+   method     setKey(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 	>   interface  com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterIconStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$KeyStep
     [exec] 		+   method     setKey(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 	+   interface  com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   access     abstract
     [exec] 		+   access     static
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$BuildStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$LabelStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$QuickActionStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$SeparatorStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$TargetStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$TypeStep
     [exec] 		+   method     build()
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItem
     [exec] 		+   method     setLabel(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterLabelStep
     [exec] 		+   method     setLabel(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterLabelStep
     [exec] 		+   method     setQuickAction(boolean)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterQuickActionStep
     [exec] 		+   method     setQuickAction(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterQuickActionStep
     [exec] 		+   method     setSeparator(boolean)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterSeparatorStep
     [exec] 		+   method     setSeparator(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterSeparatorStep
     [exec] 		+   method     setTarget(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterTargetStep
     [exec] 		+   method     setTarget(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterTargetStep
     [exec] 		+   method     setType(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterTypeStep
     [exec] 		+   method     setType(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterTypeStep
     [exec] 	>   interface  com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterPutDataStep
     [exec] 		+   implements com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$KeyStep
     [exec] 		+   method     setKey(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 	+   interface  com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$KeyStep
     [exec] 		+   access     abstract
     [exec] 		+   access     static
     [exec] 		+   method     setKey(com.liferay.petra.function.UnsafeSupplier)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] 		+   method     setKey(java.lang.String)
     [exec] 			+   access     abstract
     [exec] 			+   return     com.liferay.frontend.taglib.clay.servlet.taglib.util.DropdownItemBuilder$AfterKeyStep
     [exec] [Baseline Warning] Bundle Version Change Recommended: 10.0.0
     [exec] [Baseline Report] Mode: diff (persisted)
     [exec] Semantic versioning is incorrect while checking /opt/dev/projects/github/liferay-portal/tools/sdk/dist/com.liferay.frontend.taglib.clay-9.3.1.jar against /opt/dev/projects/github/liferay-portal/.gradle/caches/modules-2/files-2.1/com.liferay/com.liferay.frontend.taglib.clay/9.3.0/a67b90adbd7957dab7cf9ffd7592494d3cb37eeb/com.liferay.frontend.taglib.clay-9.3.0.jar
```

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
@boton @ambrinchaudhary I'm not able to reproduce the SF failure locally. Can you take a look at it?

**Comment by boton** [comment]
I going to take a look after finishing with #3108 looks easy.

**Comment by boton** [comment]
ci:test:sf

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:reevaluate:1297401_3606

**Review by ambrinchaudhary — APPROVED** [review]

**Comment by boton** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:reevaluate:1317401_4123

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:onedrive

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1317401_4251

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:reevaluate:1317401_4251

**Comment by adolfopa** [comment]
ci:reevaluate:1317401_4251

**Comment by adolfopa** [comment]
ci:reevaluate:1317401_4251

**Comment by adolfopa** [comment]
Manually forwarded to bchan: https://github.com/brianchandotcom/liferay-portal/pull/120689

**Pull request closed without merging**